lib/template-block.js

```diff
@@ -0,0 +1,32 @@
+module.exports.register = function (registry, config) {
+const handlebars = require("handlebars")
+  
+// [template,example$foo.json]
+
+  registry.block(function () {
+    var self = this
+    
+    self.named('template')
+    self.positionalAttributes('resourceId')
+    self.onContext(['listing', 'open', 'paragraph'])
+    
+    self.process(function (parent, reader,attributes) {
+      const target = config.contentCatalog.resolveResource(
+        attributes.resourceId, config.file.src)
+      
+      if (target) {
+        var data = JSON.parse(target.contents)
+        var source = reader.getLines().join('\n')
+        
+        var template = handlebars.compile(source)
+        
+        var output = template(data)
+        return self.createOpenBlock(parent, output)
+      }
+      else {
+        return self.createOpenBlock(parent, `Data file ${attributes.resourceId} not resolved`)
```

Comment:
It would be best to log an message here. Refer to these pages to learn how:

* https://docs.asciidoctor.org/asciidoctor/latest/extensions/logging/
* https://docs.asciidoctor.org/asciidoctor.js/latest/processor/logging-api/

Reply:
@mojavelinux I'm confused on how to `include Asciidoctor::Logging` in JS rather than Ruby?

The second link for JS appears to be for choosing the overall Logger for the whole processing, as opposed to calling the logger from an extension.

Reply:
Yikes, I found this on the internets, but feels like there must be a simpler way?

```
  const Opal = global.Opal
  const loggerManager = Opal.module(null, 'Asciidoctor').LoggerManager
  const logger = loggerManager.getLogger()
```

Reply:
Well, I've pushed a commit with that in meantime, but look forward to pointers to improve!

Reply:
I looked into it and the use of the Logging module in extensions is not yet well supported yet in Asciidoctor.js.

Here's the low-level JavaScript that would mix in the module:

```js
self.$singleton_class().$include(global.Opal.Asciidoctor.Logging)
```

However, that won't give you access to `getLogger()` and `createLogMessage()` because this doesn't mix in Asciidoctor.js' porcelain interface.

It's probably best to use the Logging module directly:

```js
const Logging = global.Opal.Asciidoctor.Logging

...

Logging.getLogger().error('message')

// or

Logging.getLogger().error(
  Logging.createLogMessage('message', { source_location: parent.getDocument().getReader().$cursor_at_mark() })
)
```

It's also possible to get the logger from the parent or reader:

```js
parent.getLogger()
reader.getLogger()
```

That will give you the ability to log a basic message. However, in order to use `createLogMessage` to add context, you still need to use the Logging module.